src/js/async-event-target.js

```diff
@@ -0,0 +1,33 @@
+import asyncTimers from './mixins/async-timers.js';
+import EventTarget from './event-target.js';
+
+class AsyncEventTarget extends EventTarget {
+  constructor(options = {}) {
+    super();
+    asyncTimers(this);
+
+    if (options.disposeWith) {
```

Comment:
I am unsure wether we should have something like this, but we do have automatic disposal on player dispose for a lot of our regular components. This seemed like a good idea as it allows an AsyncEventTarget to be bound to the lifecycle of something else.